### ISSUE TITLE
Addon Clashes with other CSS addons; How-to isolate or Reset?

### ISSUE THREAD

**Issue by e-karma** [bitzik/ember-cli-fullpagejs-view#2]
Hello Again,

I'm trying to utilize this addon for a landing page (from the application 'index' folder) and when its in use with the ember-ionic mobile framework addon, they clash and it prevents scrolling (position:fixed) from the other parts of the web app.

Ideally i'd love to use both, but unless I can come up with a fix I have to drop one.. Have you experienced anything like this?

Possible solutions: 

-Isolate the CSS?  If so, how? A component?
-implement a CSS reset? after the landing page?  

Thanks in Advance!

**Comment by leojpod** [comment]
yes do you have the same issue when you bypass the landing screen ? (going directly to the URL of another ember page?) 

I found out about something similar not so long ago and I made a dirty patch in my project to go round it for now, I will try wrapping it up within this addon, can you check if bypassing the landing screen works? if so then we are probably facing the same issue and I can send you an explanation on how to get the quick path working :)

**Comment by e-karma** [comment]
Thanks for the prompt reply!  Yes, the issue remains even when bypassing the landing page.  Originally I tried a home page redirect to '/landing' so the main navigation pages could start from '/' but that didn't work either and made direct routing forward to the landing for some reason.

**Comment by leojpod** [comment]
hmm ok .. do yo have a code sample somewhere that I could checkout and see if I can find anything?

**Comment by e-karma** [comment]
hmmm I could try to recreate the problem, but I just finished untangling it as I wasn't confident I'd get it working.  Do you want to share the 'patch' n i'll try that first.. if that doesn't work i'll try n recreate from the last spot?

**Comment by leojpod** [comment]
yes sure let me pull it up :)

**Comment by leojpod** [comment]
So here is what I placed within the `app/views/application.js`: 

``` javascript
    killFullPageJS: on('willInsertElement', function () {
        Ember.$.fn.fullpage.destroy();
        Ember.$.fn.fullpage.destroy('all'); 
    })
```

it's pretty ugly but I didn't had time to figure out better!
the main problem is: fullpagejs is supposed to be cleaned up entirely when leaving the view but it doesn't do it well enough when I call those methods on `willRemoveElement` within the addon. 
However, killing it after made it work...

**Comment by e-karma** [comment]
Great thanks! i'll try that out now and let you know.  I have no problem with a dirty hack for now since Embers killing the view layer soon anyway and at that point the addon will probably need to be reworked anyway...

**Comment by leojpod** [comment]
yep I'll keep an eye on that and I'll do my best to maintain this addon :)

**Comment by e-karma** [comment]
Awesome! Much appreciated

ok, I tried it and ran into this error =>
Error while processing route: index on is not defined ReferenceError: on is not defined

That snippet isn't supposed to be within the fullpageOptions is it?  Heres the view code  from the index folder..

export default FullpageJsView.extend({
  fullpageOptions: {
    sectionsColor: ['', '#151515', '', '#151515', ''],
    anchors: ['intro', 'Fast', 'Affordable', 'Quality'],
    scrollingSpeed: 800,
  },
  killFullPageJS: on('willInsertElement', function () {
     Ember.$.fn.fullpage.destroy();
     Ember.$.fn.fullpage.destroy('all');
   })
});

I also moved this all to the application folder and got the same results.

**Comment by leojpod** [comment]
No not really 
it's supposed to go directly into `app/views/application.js`. Also `on` is actually `Ember.on` I just tend to always use `cont {plenty of stuff} = Ember;` in the beginning of the my files. 

So to sum up: 
create a file `/app/views/application.js` and put : 

``` javascript
import Ember from 'ember';

export default Ember.View.extend({
  killFullpageJS: Ember.on('willInsertElement', function () {
    Ember.$.fn.fullpage.destroy();
    Ember.$.fn.fullpage.destroy('all');
  })
}
```

**Comment by e-karma** [comment]
ok, that's now working without an error, but non-landing pages are still not scrolling... position: fixed remains..

As a recap, theres 2 views ; 
-1 exactly as you showed for the application
-1 with just fullPageOptions as shown above for the index view...

I'm going to recreate a new app / post to github that reflects the issue with no added mess.  Will share it in a few

**Comment by leojpod** [comment]
ok I'll have a look at it :)

**Comment by e-karma** [comment]
Here it is... fresh 0.2.3 ember-cli app w/ the fullpageView tested first & working w/ scrolling until I added the ionic framework.  Then the scrolling for "route1" (and others if tested I imagine) is disabled. 

https://github.com/e-karma/full-ionic

Thanks again so much for the time you've spent helping, i'll make sure I can help the addon progress when I get a better grip of it!

**Comment by leojpod** [comment]
ok it's 1am at my place :P so I'll look at it in the morning!

**Comment by e-karma** [comment]
ahhhh ok, sorry didnt realize! 4 PM here :) 

gnite n thanks again! I'll be up n online working for another 12 hours or so

**Comment by leojpod** [comment]
lol no problem :)

**Comment by e-karma** [comment]
Hey man, thanks for the help, I fixed it through a diff method...  Appreciate it!

**Comment by leojpod** [comment]
Hi there! How did you fix it?

**Comment by e-karma** [comment]
Ionic is a robust 7k lines of code framework so I just pulled out a few of the main components, added them in and that must have avoided the body tag conflict.

**Comment by e-karma** [comment]
Hello again!! Sorry to be such a pain, I just really love this addon but it doesn't seem to play nice with some key ingredients of my toolbelt.   

I use PostCSS & a pretty sweet new grid framework (lost-grid), but it completely explodes the landing Fullpage CSS / experience when used together.  Do you see these as short-term issues or fundamental challenges where 'either/or' trade-off decisions need to be made when using it? 

Really trying to avoid ditching it as I love the landing-page experience / auto-scroll.  Any ideas?

**Comment by leojpod** [comment]
I think in a near future the addon should be updated to NOT load the fullpage.js css but instead inject these rules only when a fullpage.js view is being used. 
However, I don't think I'll have time for that before this summer ...

**Comment by e-karma** [comment]
Interesting solution.. I like it.  Were you thinking something along the lines of shadow dom? 
-> http://www.html5rocks.com/en/tutorials/webcomponents/shadowdom/

If you had 15 minutes in the next month to help me grok how it works in current form and maybe slip some notes into the code where helpful, i'd do my best to push it forward (Still kinda JR. tho!).  It seems likely to need an update to adapt to ember2.0 / glimmer regardless right?

**Comment by e-karma** [comment]
Good news!  I cloned a glimmer/ember2.0 styled app and moved the view from application to the index and it worked!  I tried this before but must have missed a step.   When I have some time and make it more ember2.0 compatible, i'll package it up and share.  (probably just use components)

**Comment by leojpod** [comment]
Hi @e-karma 

That would be really cool!

I have been pretty busy away from Ember this last months but I'll get back to it soon. I'll be glad to help you with that!
